### PR TITLE
Create a release process

### PR DESCRIPTION
**Pull request by Lotes**
With these changes I add a release process for Typir.
Please have a look at the `RELEASE.md` and the workflow file `publish.yml`.

The most instructions were taken and adapted from Langium. In the end any repo contributor will be able to create a release (from my current understanding).

Mind the update-script: It will be called after changing the package versions of `typir` and `typir-langium`.

What is missing at the moment is the secret NPM token in the GitHub repository. I will generate it soon. In the meanwhile this PR is in draft mode.

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+The release process is mostly automated and requires running only a few commands. After commiting, pushing, tagging and releasing the changes, a GitHub Action will publish all npm packages.
+
+1. Pull the latest changes
+2. Uplift the package versions
+   Run `npm version major|minor|patch --no-git-tag-version --workspaces`
+3. Update the dependency versions
+   Run `npm run version:dependencies`
+4. Create a PR with your updated changes, get a review and merge it
+5. Create a version tag on the latest commit on main and push it
+6. Create a GitHub release from the new tag (this will automatically publish all artifacts)
```
Comment (JohannesMeierSE) [inline RELEASE.md:12]: How to do this step? Maybe it is obvious, but I have no idea ...

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Release Process
+
+The release process is mostly automated and requires running only a few commands. After commiting, pushing, tagging and releasing the changes, a GitHub Action will publish all npm packages.
```
Comment (JohannesMeierSE) [inline RELEASE.md:3]: Is it a good idea to explicitly say, that this procedure works on any branch (and not only on `main`) or is that trivial?

File: examples/lox/package.json [hunk 3]
```diff
@@ -1,8 +1,9 @@
 {
-    "name": "lox",
+    "name": "typir-example-lox",
```
Comment (JohannesMeierSE) [inline examples/lox/package.json:2]: Thanks, that makes sense!

**Review by JohannesMeierSE — APPROVED** [review]
Looks nice for me, but please note that I have nearly no knowledge about these procedures 🙂

File: RELEASE.md (hunk 2 above)
Comment (Lotes) [reply RELEASE.md:3]: Not sure, would that not confuse future maintainers?

Comment (JohannesMeierSE) [reply RELEASE.md:3]: I don't know, let's skip it then

**Pull request merged**